rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -368,6 +368,18 @@ void Node::register_param_change_callback(CallbackT && callback)
   this->parameters_callback_ = callback;
 }
 
+template<typename ParameterT>
+bool Node::get_parameter(const std::string & name, ParameterT & parameter) const
+{
+  std::lock_guard<std::mutex> lock(mutex_);
+
+  rclcpp::parameter::ParameterVariant parameter_variant(parameter);
+  bool result = get_parameter_(name, parameter_variant);
```

Comment:
Could you not call the `bool get_parameter(std::string, ParameterVariant)` version of the function here and avoid the mutex lock two lines above?